### ISSUE TITLE
Update `Metrics/PerceivedComplexity` to reduce perceived complexity of simple pattern matching

### ISSUE THREAD

**Issue by kevinrobell-st** [rubocop/rubocop#15300]
### Before submitting

- [x] I have searched the existing issues and this hasn't been requested yet.

### Cop name

Metrics/PerceivedComplexity

### Is your feature request related to a problem?

I've noticed that a conscious choice was made to evaluate `case...in` differently from `case...when` with this cop. Each part of the `case...in` is counted the same as each part of a `if...elsif` while the `case...when` gets discounted as mentioned in the documentation to `0.2` branches.

I understand that pattern matching can be much more complex than simple case statements but I think that pattern matching weight should be evaluated based on the complexity of the expression as at times it can simplify code. It can get flagged by this rule though which indicates that the code has become more complex even though that is incorrect.

### Describe the solution you'd like

My proposal is that `case...in` branches with one literal argument should be weighted the same as `case...when` branches.

### Code examples

Specifically, I've run into the current pattern in our codebase. Note that this is simplified. Imagine that there are more cases.

```rb
case color
when "red"
  ...
when "blue"
  ...
when "green"
  ...
else
  raise ArgumentError, "unsupported color: #{color}"
end
```

I tried to replace it with this since pattern matching is strict and an error is raised on a missing branch.

```rb
case color
in "red"
  ...
in "blue"
  ...
in "green"
  ...
end
```

It gets flagged by the `Metrics/PerceivedComplexity` cop after the change.

### Describe alternatives you've considered

Using `case...when` as we did previously.
